python_modules/libraries/dagster-shell/dagster_shell/solids.py

```diff
@@ -1,51 +1,45 @@
 import os
+from enum import Enum
+from typing import Any, Dict, Optional
 
 from dagster import (
-    Enum,
-    EnumValue,
+    Config,
     Failure,
-    Field,
     In,
-    Noneable,
     Nothing,
+    OpExecutionContext,
     Out,
-    Permissive,
     _check as check,
     op,
 )
+from pydantic import Field as PyField
 
 from .utils import execute, execute_script_file
 
 
-def shell_op_config():
-    return {
-        "env": Field(
-            Noneable(Permissive()),
-            is_required=False,
-            description="An optional dict of environment variables to pass to the subprocess.",
-        ),
-        "output_logging": Field(
-            Enum(
-                name="OutputType",
-                enum_values=[
-                    EnumValue("STREAM", description="Stream script stdout/stderr."),
-                    EnumValue(
-                        "BUFFER",
-                        description="Buffer shell script stdout/stderr, then log upon completion.",
-                    ),
-                    EnumValue("NONE", description="No logging"),
-                ],
-            ),
-            is_required=False,
-            default_value="BUFFER",
-        ),
-        "cwd": Field(
-            Noneable(str),
-            default_value=None,
-            is_required=False,
-            description="Working directory in which to execute shell script",
-        ),
-    }
+class OutputType(Enum):
+    STREAM = "STREAM"
+    BUFFER = "BUFFER"
+    NONE = "NONE"
+
```

Comment:
what happens to the enum descriptions?

Reply:
I'm not sure what the best way to model this would be. In an ideal world we could use attribute docstrings, but those are not available at runtime in Python:
```python
class OutputType(Enum):
    STREAM = "STREAM"
    """Stream script stdout/stderr.
    """
```

Other options are an optional ABC that lets users define a description function, or maybe something on the `Field` which points at the enum class.

Reply:
So at a minimum preserve the existing descriptions as comments